scripts/data_processing/batch_grid_analysis.py

```
@@ -0,0 +1,433 @@
+
+    def run_grid_analysis(self, image_path: Path) -> Optional[Dict[str, Any]]:
+        """Run grid detection comparison on an image"""
+        try:
+            print(f"Analyzing: {image_path.name}")
+
+            # Set environment for proper Unicode handling
+            env = os.environ.copy()
+            env["PYTHONIOENCODING"] = "utf-8"
+
+            # Run the comparison script
+            result = subprocess.run(
```

Ruff (0.12.2)

`subprocess` call: check for execution of untrusted input

(S603)